scripts/which_clang_files

```diff
@@ -0,0 +1,33 @@
+#! /bin/bash
+
+HERE="$( dirname -- "${0}"; )"
+WAKE_ROOT="${HERE}/../"
+
+if [[ ${1} == "all" ]]; then
+  SOURCE_FILES=$(find ${WAKE_ROOT} -type f \( -name "*.h" -o -name "*.cpp" -o -name "*.c" \) | grep -v 'vendor')
+
+  echo $SOURCE_FILES
+
+  # Flag if any files were found
+  if [ "$SOURCE_FILES" ]; then
+    exit 0
+  fi
+  exit 1
+fi
+
+if [[ ${1} == "changed" ]]; then
+  CHANGED_FILES=$(git --git-dir ${WAKE_ROOT}.git diff --name-only | grep '\.h\|\.cpp\|\.c' | grep -v 'vendor/')
+  STAGED_FILES=$(git --git-dir ${WAKE_ROOT}.git diff --name-only --cached | grep '\.h\|\.cpp\|\.c' | grep -v 'vendor/')
+  MERGED="${CHANGED_FILES} ${STAGED_FILES}"
+  
+  echo ${MERGED}
+
+  # Flag if any files were found
+  if [ "${MERGED}" ]; then
+    exit 0
+  fi
+  exit 1
+fi
+
+echo "unknown parameter: '${1}'"
+exit 1
```

Comment:
BTW, for whatever reason everyone else that works on wake has their vscode configured to always insert a newline at the end of the file. Consequently if they open a file to view it and it *doesn't* have a newline their editor adds it. Consequently we're all forced to do it 🤷 

TL;DR: Always have a newline at the end of your files. You can add this as a setting in vscode....why on earth we chose this convention I don't know.

Reply:
A line of text in posix is defined as having an newline at the end, so some tools might expect it and may ignore the last "line" if it's just EOF rather than \nEOF

Reply:
Maybe its just strange that this isn't the default in vscode?